src/api/list_drop.sql

```diff
@@ -0,0 +1,64 @@
+DROP FUNCTION IF EXISTS api.list_drop(jsonb);
+
+CREATE OR REPLACE FUNCTION api.list_drop(data jsonb)
+RETURNS void AS $$
+DECLARE  
+	class          jsonb;
+	obj_id 		   uuid;
+	value_to_drop  jsonb; 
+	new_value 	   jsonb;
+	json_path      text[];
+	obj		 	   jsonb;
+	new_obj		   jsonb;
+	field_value    jsonb;
+	access_token   jsonb;
+
+	BEGIN
+		class := data->'class';
+		IF (class IS NULL) THEN
+			RAISE EXCEPTION 'The reclada object class is not specified';
+   		END IF;
+	
+   	
+		obj_id := (data->>'id')::uuid;
+		IF(obj_id IS NULL) THEN
+        	RAISE EXCEPTION 'The is no id';
+    	END IF;
+	
+    	access_token := data->'access_token';
+		SELECT api.reclada_object_list(format(
+			'{"class": %s, "attrs": {}, "id": "%s", "access_token": %s}',
+        	class,
+        	obj_id,
+        	access_token
+    		)::jsonb) -> 0 INTO obj;
+    	
+    	IF (obj IS NULL) THEN
+        	RAISE EXCEPTION 'The is no object with such id';
+    	END IF;
+    	
+    	
+    	json_path := format('{attrs, %s}', data->'field');
+    
+		value_to_drop := data->'value';
+		IF (value_to_drop IS NULL) THEN
+        	RAISE EXCEPTION 'There value should not be null';
+    	END IF;
+    	
+    	field_value := obj#>json_path;
+		    
+    	SELECT jsonb_agg(elems)
+		FROM 
+			jsonb_array_elements(field_value) elems
+		WHERE 
+			NOT (elems IN ( 
+				SELECT jsonb_array_elements(value_to_drop)))
+		INTO new_value;
```

Comment:
There is a way to remove array items without complex selects
```
# select '["a", "b"]'::jsonb - 'b';
 ?column? 
----------
 ["a"]
(1 row)
```

However, value_to_drop is jsonb, and in order use '-' it should be converted to scalar; the full expression being:

`field_value - (value_to_drop >> {})`

Reply:
Didn't change it because value_to_drop (now:values_to_drop) is array